### PR TITLE
EL-1745 Check Answers display

### PR DESCRIPTION
**Pull request by willc-work**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1745)

## What changed and why

Added a check to see if we are responding to an EE gross income result
If we are we display only relevant sections on Check ANswers

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: app/services/check_answers/section_lister_service.rb [hunk 1]
```diff
@@ -28,13 +28,15 @@ def call(session_data)
               Sections::Dependants.new(check),
               Sections::ClientIncome.new(check),
             ]
-            post = [Sections::Outgoings.new(check)]
+            post = check.gross_early_eligibility_exit? ? [] : [Sections::Outgoings.new(check)]
             finance_sections = if check.partner?
                                  pre + [Sections::PartnerIncome.new(check)] + post
```
Comment (MazOneTwoOne) [inline app/services/check_answers/section_lister_service.rb:33]: I think we also have to put in a ternary operator here, for the partner income i.e

`pre + (check.gross_early_eligibility_exit? ? [] : [Sections::PartnerIncome.new(check)]) + post`

We are showing these screens even though user did not fill those questions in.

**Review by MazOneTwoOne — COMMENTED** [review]
Just the one query. 

Also, do we need to put in a bunch of flow tests?

File: app/services/check_answers/section_lister_service.rb (hunk 1 above)
Comment (willc-work) [reply app/services/check_answers/section_lister_service.rb:33]: thanks good spot, I think I was just expecting to see income questions.

**Comment by willc-work** [comment]
@MazOneTwoOne re [this comment](https://github.com/ministryofjustice/laa-check-client-qualifies/pull/1550#pullrequestreview-2318622289) I am not keen on the idea of adding a bunch of tests to check what displays given that UATing it will confirm that for us without the ongoing overhead of running 3 or 4 tests to cover the variations needed for this bug. If the bug was more serious then tests may be required.

**Comment by MazOneTwoOne** [comment]
> @MazOneTwoOne re [this comment](https://github.com/ministryofjustice/laa-check-client-qualifies/pull/1550#pullrequestreview-2318622289) I am not keen on the idea of adding a bunch of tests to check what displays given that UATing it will confirm that for us without the ongoing overhead of running 3 or 4 tests to cover the variations needed for this bug. If the bug was more serious then tests may be required.

Maybe not a bunch of tests, but at least we should update this spec `spec/flows/early_result_spec.rb` to cover the new behaviour (technically the existing behaviour we should have had)?

It may stop us adding in a similar regression.

**Review by MazOneTwoOne — APPROVED** [review]

File: app/services/check_answers/section_lister_service.rb [hunk 2]
```diff
@@ -26,15 +26,22 @@ def call(session_data)
...
             pre = [
               Sections::Dependants.new(check),
-              Sections::ClientIncome.new(check),
-            ]
-            post = [Sections::Outgoings.new(check)]
+              # Simplecov is showing that check.under_eighteen_assets? is not being covered by a test
+              # we have tests in spec/flows/under_eighteen_flow_spec.rb that check with and without under_eighteen_assets
+              # :nocov:
+              (Sections::ClientIncome.new(check) if !check.under_eighteen? || check.aggregated_means? || check.under_eighteen_regular_income? || check.under_eighteen_assets?),
+              # :nocov:
```
Comment (MazOneTwoOne) [inline app/services/check_answers/section_lister_service.rb:33]: @willc-work this feels risky 🤔

File: spec/flows/early_result_spec.rb [hunk 3]
```diff
@@ -32,8 +32,47 @@
...
+    #            "Type of matter",
+    #            "Type of immigration or asylum matter",
+    #            "Number of dependants",
+    #            "Employment status",
+    #            "Client employment income 1",
+    #            "Client benefits",
+    #            "Client other income"],
+    #           )
+    # end
+
```
Comment (MazOneTwoOne) [inline spec/flows/early_result_spec.rb:75]: @willc-work did we mean to leave this commented out?

File: app/services/check_answers/section_lister_service.rb (hunk 2 above)
Comment (willc-work) [reply app/services/check_answers/section_lister_service.rb:33]: we have tests that cover both sides of under 18 assets and the associated views, so I think i have mitigated the risk as well as possible without spending more time understanding why simplecov isnt counting this as coverage

File: spec/flows/early_result_spec.rb (hunk 3 above)
Comment (willc-work) [reply spec/flows/early_result_spec.rb]: removed now., this was to test if adding income values triggered the coverage, it doesnt

File: app/services/check_answers/section_lister_service.rb (hunk 2 above)
Comment (MazOneTwoOne) [reply app/services/check_answers/section_lister_service.rb:33]: Happy to leave like this. But maybe we make a separate ticket for this, so this doesn't get lost?

**Pull request merged**